src/SingleArticle.jsx

```diff
@@ -1,32 +1,51 @@
 import { useParams } from "react-router-dom";
 import { getSingleArticle } from "./Api";
 import { useState, useEffect } from "react";
+import Comments from "./Comments";
 
 function SingleArticle() {
     const { article_id } = useParams();
-    const [singleArticle, setSingleArticle] = useState ([])
+    const [singleArticle, setSingleArticle] = useState ([]);
+    const [isLoading, setIsLoading] = useState(true);
+    const [isError, setIsError] = useState(null)
```

Comment:
Because the `isError` variable is being set with the `err` object that comes back from the catch, it might be best to initialise the state with `{}` instead of `null` so that it's always the same data type